scripts/us_census/acs5yr/subject_tables/jsonSpec.md

```diff
@@ -0,0 +1,203 @@
+			"dependentPVs": {
+				"armedForcesStatus": "Civilian",
+				"institutionalization": "USC_NonInstitutionalized"
+			}
+		}
+   ]
+```
+
+
+#### `inferedSpec` [optional]
+Adds additional property-value pairs to a stat-var node based on the inference of an existing property in that stat-var node.
+
+For instance, in the example, if the stat-var node has a property `employment` defined, additional property-values for `employmentStatus` and `age` are added to the stat-var node.
+
+```json
+  "inferredSpec":{
+		"employmentStatus": {
+			"age": "[16 - Years]"
+		},
+		"employment": {
+			"employmentStatus": "BLS_InLaborForce",
+			"age": "[16 - Years]"
+		}
+   }
+```
+
+#### `enumSpecializations` [optional]
+This section is not used in the stat-var node but is used to process the column name for which a stat-var node is to be generated.
+
+
+```json
+  "enumSpecializations":{
+		"Under 6 years": "Under 18 years, Under 19 years",
+		"6 to 17 years": "Under 18 years",
+		"6 to 18 years": "Under 19 years",
+		"18 to 24 years": "18 to 64 years",
+		"24 to 34 years": "18 to 64 years",
+		"19 to 25 years": "19 to 64 years",
+		"26 to 34 years": "19 to 64 years",
+		"35 to 44 years": "18 to 64 years, 19 to 64 years",
+		"45 to 54 years": "18 to 64 years, 19 to 64 years",
+		"55 to 64 years": "18 to 64 years, 19 to 64 years",
+		"65 to 75 years": "65 years and older"
+   }
+```
+For instance, if the column name is *Estimate!!Total Uninsured!!Total civilian noninstitutionalized population!!AGE!!Under 19 years!!Under 6 years*, the column token *Under 6 years* is a specialization of *Under 19 years*. So, while generating the stat-var node for this column we would like to keep only the specialization, which will mean the stat-var node generated will be for the column *Estimate!!Total Uninsured!!Total civilian noninstitutionalized population!!AGE!!Under 6 years*
+
+
+#### `ignoreColumns` [optional]
+This section of the spec specifies a list of tokens that if present in the column name, no stat-var node is generated for that column and the column is ignored.
+
+```json
+	"ignoreColumns": [
+		"id",
+		"Geographic Area Name",
+		"Total",
+		"Total MOE"
+	]
+```
+
+### Customizations and extensibility
+The JSON spec can be further extended and customized for each subject tables. At the moment, we support the following extensions.
+
+#### `denominators` [optional]
+If the subject table has percentage values, and you are interested to convert them to numerical counts, then the columns to map can be specified in the denominators section.
+For instance `Uninsured Population!!Estimate!!WORK EXPERIENCE!!Civilian noninstitutionalized population 16 to 64 years` represents the total count of population between 16 and 64 years and the column `Uninsured Population!!Estimate!!Worked full-time, year round in the past 12 months` represents a fraction of the total count of population between 16 and 64 years who worked full-time, year round in the past 12 months.
+
+```json
```

Comment:
pls add a note in the comment if these have to be full column names or can they be tokens as well?